src/mlpack/methods/time_series_models/util/lag.hpp

```diff
@@ -0,0 +1,59 @@
+/**
+ * @file mothods/time_series_models/util/lag.hpp
+ * @author Rishabh Garg
+ *
+ * Implementation of Lag() which returns the feature shifted by a specified
+ * period.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_TIME_SERIES_MODELS_UTIL_LAG_HPP
+#define MLPACK_METHODS_TIME_SERIES_MODELS_UTIL_LAG_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core.hpp>
+
+namespace mlpack {
+namespace ts /* Time Series methods. */ {
+
+/**
+ * It returns the lagged feature of the input time series vector.
+ *
+ * @tparam InputType Type of input (arma::rowvec or arma::colvec).
+ * @param input Input data to create lagged feature.
+ * @param k Number of periods to shift (must be a non negative integer).
+ */
+template<typename InputType>
+InputType Lag(const InputType& input, size_t k) {
```

Comment:
I guess we could use http://arma.sourceforge.net/docs.html#shift instead but on the other side I'm wondering if it makes sense to do the shift inplace. What do you think?

Reply:
From memory point of view, you are right but a practical use case for the lag function will be to create lagged features and ideally, we would require our original feature and also our lag feature. 

For example, let's take the case of persistence model. In this, we are using the lagged feature for making the predictions Now while making predictions, we don't want to loose our actual labels. So, thats why I believe we need another object to store the output of this function. I hope the example clarifies the point I am trying to convey.